src/tests/plans/D3MAaveV3Plan.t.sol

```diff
@@ -0,0 +1,230 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import { Hevm, D3MPlanBaseTest } from "./D3MPlanBase.t.sol";
+import { DaiLike, TokenLike } from "../interfaces/interfaces.sol";
+
+import { D3MAaveV3Plan, PoolLike } from "../../plans/D3MAaveV3Plan.sol";
+
+interface InterestRateStrategyLike {
+    function getBaseVariableBorrowRate() external view returns (uint256);
+    function getMaxVariableBorrowRate() external view returns (uint256);
+    function calculateInterestRates(
+        address reserve,
+        uint256 availableLiquidity,
+        uint256 totalStableDebt,
+        uint256 totalVariableDebt,
+        uint256 averageStableBorrowRate,
+        uint256 reserveFactor
+    ) external returns (
+        uint256,
+        uint256,
+        uint256
+    );
+}
+
+contract D3MAaveV3PlanWrapper is D3MAaveV3Plan {
+
+    constructor(address dai_, address pool_) D3MAaveV3Plan(dai_, pool_) {}
+
+    function calculateTargetSupply(uint256 targetInterestRate) external view returns (uint256) {
+        uint256 totalDebt = stableDebt.totalSupply() + variableDebt.totalSupply();
+        return _calculateTargetSupply(targetInterestRate, totalDebt);
+    }
+}
+
+contract D3MAaveV3PlanTest is D3MPlanBaseTest {
+    PoolLike aavePool;
+    InterestRateStrategyLike interestStrategy;
+    TokenLike adai;
+
+    // Allow for a 1 BPS margin of error on interest rates
+    uint256 constant INTEREST_RATE_TOLERANCE = RAY / 10000;
+
+    function setUp() public override {
+        contractName = "D3MAaveV3Plan";
+
+        dai = DaiLike(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+        aavePool = PoolLike(0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2);
+        adai = TokenLike(0x018008bfb33d285247A21d44E50697654f754e63);
+        interestStrategy = InterestRateStrategyLike(0x694d4cFdaeE639239df949b6E24Ff8576A00d1f2);
+
+        d3mTestPlan = address(new D3MAaveV3PlanWrapper(address(dai), address(aavePool)));
+    }
+
+    function assertEqInterest(uint256 _a, uint256 _b) internal {
+        uint256 a = _a;
+        uint256 b = _b;
+        if (a < b) {
+            uint256 tmp = a;
+            a = b;
+            b = tmp;
+        }
+        if (a - b > INTEREST_RATE_TOLERANCE) {
+            emit log_bytes32("Error: Wrong `uint' value");
+            emit log_named_decimal_uint("  Expected", _b, 27);
+            emit log_named_decimal_uint("    Actual", _a, 27);
+            fail();
+        }
+    }
+
+    function test_sets_adai() public {
+        assertEq(address(adai), D3MAaveV3Plan(d3mTestPlan).adai());
+    }
+
+    function test_sets_adaiRevision_value() public {
+        assertEq(D3MAaveV3Plan(d3mTestPlan).adaiRevision(), 1);
```

Comment:
Tbh a comment above this test or assert would be sufficient but otherwise we do have historical record (this comment) if we don't want to put it in the code